nbconvert/exporters/script.py

```diff
@@ -19,15 +18,29 @@ def _template_file_default(self):
 
     def from_notebook_node(self, nb, resources=None, **kw):
         langinfo = nb.metadata.get('language_info', {})
-        
         # delegate to custom exporter, if specified
         exporter_name = langinfo.get('nbconvert_exporter')
         if exporter_name and exporter_name != 'script':
             self.log.debug("Loading script exporter: %s", exporter_name)
-            from .export import get_exporter
+            from .export import get_exporter, ExporterNameError
             if exporter_name not in self._exporters:
-                Exporter = get_exporter(exporter_name)
-                self._exporters[exporter_name] = Exporter(parent=self)
+                try: 
+                    Exporter = get_exporter(exporter_name)
+                    self._exporters[exporter_name] = Exporter(parent=self)
+                except ValueError:
+                    if langinfo.get('nbconvert_exporter_package',""):
+                        external_exporter_name = langinfo.get('nbconvert_exporter_package')
+                        try: 
+                            Exporter = get_exporter(external_exporter_name)
```

Comment:
Embedding data in the notebook like that is definitely ugly, but we've generally lived with it as a compromise so we can view/convert notebooks without the kernel in question installed.

I agree that the entry points approach is neater, and it reduces the security concern over recommending installing a third party package. But let's take a couple of days to mull it over before anyone starts implementing, because I feel like I'm getting tunnel vision on this.